src/web/mjs/connectors/Allanimesite.mjs

```diff
@@ -0,0 +1,106 @@
+    }
+    async _getMangaFromURI(uri) {
+        const request = new Request(new URL(uri), this.requestOptions);
+        const data = await this.fetchDOM(request, this.queryMangaTitleFromURI);
+        const id = uri.pathname.replace(/$\//, '');
+        const title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
+    async _getMangas() {
+        const mangaList = [];
+        for(let page = 1, run = true; run; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            await this.wait(this.config.throttle.value);
+            mangas.length ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+    async _getMangasFromPage(page) {
+        const uri = new URL(this.path + this.varQueryMangas.replace('%PAGE%', page), this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        if (data.data == null) {
+            return {};
+        }
+        return data.data.mangas.edges.map(element => {
+            return {
+                id: '/manga/'+element._id,
+                title: element.englishName ? element.englishName : element.name,
+            };
+        });
+    }
+    async _getChapters(manga) {
+        let mangaid = manga.id.replace('/manga/', '');
+        let uri = new URL(this.path+ this.varQueryChapters.replace('%MANGAID%', mangaid), this.url);
+        let request = new Request(uri, this.requestOptions);
+        const data = await this.fetchJSON(request);
+        let chapterlist = [];
+        let subchapters = data.data.manga.availableChaptersDetail.sub;
+        subchapters.forEach(chapter => {
+            chapterlist.push({
+                id : '/read/'+mangaid+'/chapter-'+chapter+'-sub',
+                title : 'Chapter '+ chapter+' [SUB]',
+                language : 'SUB',
+            });
+        });
+        let rawchapters = data.data.manga.availableChaptersDetail.raw;
+        rawchapters.forEach(chapter => {
+            chapterlist.push({
+                id : '/read/'+mangaid+'/chapter-'+chapter+'-raw',
+                title : 'Chapter '+ chapter +' [RAW]',
+                language : 'RAW',
+            });
+        });
+        return chapterlist;
+    }
+    async _getPages(chapter) {
+        const request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        let script = `
+        new Promise(resolve => {
+            let pages = __NUXT__;
+            resolve(pages);
+        });
+        `;
+        let data = await Engine.Request.fetchUI(request, script);
+        let sourcename = data.fetch[0].selectedSourceName;
+        let sourcesArray = data.fetch[0].chapters;
+        let goodSource = null;
+        sourcesArray.every(source => {
+            if (source.sourceName == sourcename) {
+                goodSource = source;
+                return false;
+            }
+            return true;

```

Comment:
You should use strict equality (===)
Also, why are you using `every ` if you're not using the result ? Are you just trying to find the `sourcesArray.findLast()` ?
it could be simplified to `goodSource = sourcesArray.findLast(source => source.sourceName ==== sourcename);`